Self signed - Working instance with custom domain & self-signed certs

slloyd-src opened the pull request:
Curl requests now work from the command line and via the browser using the _https://src-data-repo.co.uk/torkeep_ url.

1. Requires custom domain & self-signed certs (supplied) setting up.
2. Requires being a member of _ivo://skao.int/gms?prototyping-groups/mini-src/platform-users_ on the [SKA IAM](https://ska-iam.stfc.ac.uk/login)(can be changed in baldur.properties)

See the readme.md for a detailed explanation of requests.

slloyd-src committed:
Added some self-signed certs and RA for inter-component comms (not used yet)

Updated each component's config files to use a custom URL

slloyd-src committed:
As keys are intended as a temp solution, which will be in a docker image(s), I've added them to a compressed file.

slloyd-src committed:
- nginx added for common interface

- update the reg <-> reg... reference
- injected rootCA into each component so that they can accept self-signed cert
- Added CADC's example image version script (if needed in future) will parse response from harbor (see details.md)
- Added self-signed certificates, can be replace if/when required.

slloyd-src committed:
-removed runtime only files from individual components

- Added missing tomcat user property to torkeep component

slloyd-src committed:
- Fixed httpS omission

- Removed explicit SSL types from nginx
- Changed default database name to 'metadata' in both postgres & torkeep config files
- Changed the GMS after conversation with Patrick Dowler at CADC (still to be tested - API token generation etc.)
- Separated the dbase docker-compose from the rest of the services (dbase can now be stateful if required AND dbase needs to be setup before starting the torkeep service)
- Updated the 'reg' component to use a new image (bug fix for the availability API)

slloyd-src committed:
Added a real world group to group access (individual will need to add the group in SKAO IAM system - README coming soon)

Removed the postgresDB from the docker-compose to allow for persistence (if required) and to allow it to establish itself before deploying torkeep (issue with CADC's dev postgres image not fully completing (if torkeep waits for it to complete before starting, it'll wait forever)

slloyd-src committed:
Removed temp files from git

slloyd-src committed:
- Partial readme

slloyd-src committed:
Updated README

slloyd-src committed:
Added a link to an existing repo for reference

slloyd-src committed:
custom domain details added

slloyd-src committed:
-added group for auth testing

- added Patrick's test Observation

slloyd-src committed:
- Required to use caom: prefix

- Added dev baldur permissions version
- Enabled ID manager
- have to map the host's "localhost" to our custom domain, so it is accessible from within containers

slloyd-src committed:
- baldur defined groups, singular key (readOnlyGroup, etc)

- Added HOST_IP so that each component can access other components via the nginx reverse proxy
- start shell script to allow the determination of the docker HOST_IP
- Added an extra test data file

slloyd-src committed:
updated documentation slightly to reflect new process to start services

slloyd-src committed:
A few more bits of info/clarity for usage.

slloyd-src committed:
added required permissions group

slloyd-src committed:
Formatted some text in README

pahjbo commented:
it does not "just work" for me on my mac - which is ironic - partially because the start-services.sh does not work as is on MacOS - but anyway I had a go at setting that manually - however things still do not work - this is where I am a little unsure of the intention as when I look at docker all torkeep, baldur etc. are exposed on localhost ports - but there is no exposed port from nginx, which was supposed to be doing the https proxying for the others - we perhaps need a chat

pahjbo committed:
set up the ssl proxy with nginx-proxy instead

this does work fine on mac - however it does need access to host docker socket - this might be in different places on different OSs....

pahjbo committed:
remove erroneous virtual host on nginx itself

pahjbo committed:
don't need the start services any more

slloyd-src committed:
- Removed unused properties

- Moved IAM, group & bear token details to detail.md

slloyd-src committed:
formatting adjustments

link to details.md for iam reference

slloyd-src committed:
formatting issue

slloyd-src committed:
.

slloyd-src committed:
.

slloyd-src committed:
indexing update

slloyd-src committed:
github markdown issues

slloyd-src committed:
github markdown issues

pahjbo reviewed (approved):
I think that this is 'working' - it is a shame about having to set up self signed certs - however, lets call this done now - will see if the instructions are good enough for people to use
